Explainer/process_presentation.py

```diff
@@ -0,0 +1,53 @@
+import os
+import asyncio
+
+from pptx import Presentation
+from Explainer import read_from_powerpoint, json_utils, chat_gpt_interface
+
+
+UPLOADS_FOLDER = os.getenv('UPLOADS_FOLDER_PATH')
+
+OUTPUTS_FOLDER = os.getenv('OUTPUTS_FOLDER_PATH')
+
+async def process_slide(slides):
+    UPLOADS_FOLDER = os.environ.get("UPLOADS_FOLDER_PATH")
+    tasks = await asyncio.gather (*[asyncio.create_task(chat_gpt_interface.send_query (slide))for slide in slides])
+    return tasks
+
+async def process_presentation(file_path):
+    file_path = "C:/Users/shire/Downloads/le_chien.pptx"
+    presentation = Presentation(file_path)
+    powerpoint_content = read_from_powerpoint.parse_all_power_point(file_path)
+    response = asyncio.run(process_slide(powerpoint_content))
+    json_utils.save_to_json(response, "res.json")
+    return response
+
+
+
+async def process_new_files():
```

Comment:
"Separate code to small functions-
https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be
"